### ISSUE TITLE
bluemix kitura-start demo crash

### ISSUE THREAD

**Issue by jkdntc** [IBM-Swift/Kitura#906]
I use wrk to test then crash and get a bad performance.
ps: swift version 
$ swift --version
Apple Swift version 3.0.1 (swiftlang-800.0.58.6 clang-800.0.42.1)
Target: x86_64-apple-macosx10.9

$wrk -c1000 -d5s -t50  http://localhost:8090/hello
Running 5s test @ http://localhost:8090/hello
  50 threads and 1000 connections
  Thread Stats   Avg      Stdev     Max   +/- Stdev
    Latency    93.37ms  113.72ms   1.98s    98.91%
    Req/Sec    62.55     75.92   584.00     88.25%
  10466 requests in 5.10s, 1.99MB read
  Socket errors: connect 0, read 544, write 27, timeout 94
Requests/sec:   2051.34
Transfer/sec:    398.65KB

$./.build/debug/Kitura-Starter
 INFO: Kitura_Starter main.swift line 29 - Server will be started on 'http://localhost:8090'. 
 INFO: listen(on:) HTTPServer.swift line 73 - Listening on port 8090 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 227) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 54) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 56) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 31) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 27) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 50) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 41) failed. Error = Error code: -9971, Connection reset by peer. 
 ERROR: handleRead() IncomingSocketHandler.swift line 108 - Read from socket (file descriptor 39) failed. Error = Error code: -9971, Connection reset by peer. 
Trace/BPT trap: 5

**Comment by shmuelk** [comment]
@jkdntc I have recreated your problem using the Bluemix Kitura-Starter app running locally on my mac. This is using Kitura 1.2.x

I also managed to get the code to compile using Kitura 1.3 and couldn't get the failure to occur.

I looked into the place where the code was failing using lldb. Under lldb, I got a message about an unwrapped optional. In one of the commits between KituraNet 1.2.3 and KituraNet 1.3.0 there was a fix in that area.

I can help you get the Bluiemix Kitura-Sample app to compile using Kitura 1.3, but that won't help you if you want to run it on Bluemix.

I will reach out to the appropriate people in Swift@IBM to get Bluemix-Kitura-Sample upgraded.
